app/views/shared/_client_filters.erb

```diff
@@ -0,0 +1,26 @@
+<div class="client-filters">
+  <%= form_with method: "get", local: true, builder: VitaMinFormBuilder, id: "filter-form" do |f| %>
+    <h3>Filters</h3>
+    <hr/>
+    <label for="status">Stage/Status</label>
+    <div class="select">
+      <select name="status" class="select__element" id="status-filter">
+        <option selected value></option>
+        <% TaxReturn.grouped_statuses.each do |stage, statuses| %>
+          <option value="<%= stage %>" <%= params[:status] == stage && "selected" %>><%= TaxReturnStatusHelper.stage_translation(stage) %></option>
+          <% statuses.each do |status| %>
+            <option value="<%= status %>" <%= params[:status] == status && "selected" %>>&emsp;<%= status_translation(status) %></option>
+          <% end %>
+        <% end %>
+      </select>
+    </div>
+    <%= f.hidden_field :order, value: @sort_order %>
```

Comment:
Passing in the sort order and column as hidden fields allows us to maintain the current order